cmake/AddInputFileTests.cmake

```diff
@@ -33,10 +35,23 @@ function(add_single_input_file_test INPUT_FILE EXECUTABLE CHECK_TYPE TIMEOUT)
       --check-options --input-file ${INPUT_FILE}
       )
   elseif("${CHECK_TYPE}" STREQUAL "execute")
+    # Write command into shell script to make it easier to chain multiple
+    # commands
+    file(
+      WRITE
+      ${PROJECT_BINARY_DIR}/tmp/${CTEST_NAME}.sh
+      "\
+#!/bin/sh\n\
+${CMAKE_BINARY_DIR}/bin/${EXECUTABLE} --input-file ${INPUT_FILE}\n\
+exit_code=$?\n\
+${PYTHON_EXECUTABLE} ${CMAKE_SOURCE_DIR}/tools/CleanOutput.py -v \
+--input-file ${INPUT_FILE} --output-dir ${CMAKE_BINARY_DIR}\n\
+(( exit_code = $exit_code || $? ))\n\
+exit $exit_code\n"
+    )
     add_test(
       NAME "${CTEST_NAME}"
-      COMMAND ${CMAKE_BINARY_DIR}/bin/${EXECUTABLE}
-      --input-file ${INPUT_FILE}
+      COMMAND sh ${PROJECT_BINARY_DIR}/tmp/${CTEST_NAME}.sh
```

Comment:
Then here `sh ${PROJECT_BINARY_DIR}/tmp/${CTEST_NAME}.sh ${EXECUTABLE} ${INPUT_FILE}`

Reply:
- @nilsdeppe [nitpick] Shouldn't the new command use the name of the general script for all tests (as you requested above), rather than `${PROJECT_BINARY_DIR}/tmp/${CTEST_NAME}.sh`?
- @nilsleiffischer What is the reason to write a two-line python script vs. using two CMake `COMMAND` statements (one to run the test, one to clean the executable)?

Reply:
- Yes, my mistake. Thanks for catching that :+1:
- I would agree that this is a simple enough regex that it should be possible to do in CMake. The advantage would be not having to fire up the python interpreter during test run time (python interpreter startup is horrendous), though I'm not sure it's necessary.

Reply:
@fmahebert I think you are asking about why I write the two-line _shell_ script instead of two CMake `COMMAND`s. And the answer is that `add_test` does not support multiple `COMMAND`s, and also that it is a pain to combine commands and their exit codes in CMake. I want the test to fail if either of the two commands fail, but also I want to run the cleanup to get rid of the output files even if the run failed.
@nilsdeppe About doing the cleanup in CMake instead of Python: same argument about coding in CMake is a pain, the startup time is negligible for the 10-100 input file tests we‘ll have, and also it could be useful to have the Python script available outside of CMake.